src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -486,6 +486,9 @@ public static SyntaxTree ParseWithRoundTripCheck(string text, CSharpParseOptions
             ? ImmutableArray.Create<MetadataReference>(NetStandard20.NetStandard, NetStandard20.MscorlibRef, NetStandard20.SystemRuntimeRef, NetStandard20.SystemDynamicRuntimeRef)
             : ImmutableArray.Create(MscorlibRef);
 
+        // Careful! Make sure everything in s_desktopRefsToRemove is constructed with
+        // the same object identity, since MetadataReference uses reference equality.
+        // this may mean adding Interlocked calls in the construction of the reference.
```

Comment:
This is very subtle and far from s_systemCoreRef and s_systemRef declarations. I think same comment should be added there too.